controllers/lumigo_controller_suite_test.go

```diff
@@ -318,11 +303,283 @@ var _ = Context("Lumigo controller", func() {
+			})
+
+			By("Validating deployment got injected", func() {
+				deployment := &appsv1.Deployment{}
+				Expect(k8sClient.Get(ctx, types.NamespacedName{
+					Namespace: namespaceName,
+					Name:      deploymentName,
+				}, deployment)).To(Succeed())
+
+				Expect(deployment).To(mutation.BeInstrumentedWithLumigo(lumigoOperatorVersion, lumigoInjectorImage, telemetryProxyOtlpServiceUrl))
+			})
+
+			By("Deleting the Lumigo resource", func() {
+				Expect(k8sClient.Delete(ctx, lumigo)).Should(Succeed())
+
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(ctx, types.NamespacedName{
+						Namespace: namespaceName,
+						Name:      lumigoName,
+					}, &operatorv1alpha1.Lumigo{})).To(MatchError(ContainSubstring("not found")))
+				}).Should(Succeed())
+			})
+
+			By("Validating the deployment still has injection", func() {
+				deployment := &appsv1.Deployment{}
+				Expect(k8sClient.Get(ctx, types.NamespacedName{
+					Namespace: namespaceName,
+					Name:      deploymentName,
+				}, deployment)).To(Succeed())
+
+				Expect(deployment).To(mutation.BeInstrumentedWithLumigo(lumigoOperatorVersion, lumigoInjectorImage, telemetryProxyOtlpServiceUrl))
+			})
+		})
+
+		It("should undo injection when removing the Lumigo resource", func() {
+			lumigoSecretName := "lumigo-credentials"
+			expectedTokenKey := "token"
+
+			By("Inititalizing the secret", func() {
+				Expect(k8sClient.Create(ctx, &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: namespaceName,
+						Name:      lumigoSecretName,
+					},
+					Data: map[string][]byte{
+						expectedTokenKey: []byte("t_1234567890123456789AB"),
+					},
+				})).Should(Succeed())
+			})
+
+			deploymentName := "test-deployment"
+
+			By("Inititalizing the deployment", func() {
+				deployment := &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      deploymentName,
+						Namespace: namespaceName,
+					},
+					Spec: appsv1.DeploymentSpec{
+						Selector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"deployment": deploymentName,
+							},
+						},
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{
+								Labels: map[string]string{
+									"deployment": deploymentName,
+								},
+							},
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:  "myapp",
+										Image: "busybox",
+									},
+								},
+							},
+						},
+					},
+				}
+				Expect(k8sClient.Create(ctx, deployment)).Should(Succeed())
+			})
+
+			var lumigo *operatorv1alpha1.Lumigo
+			By("Initializing the Lumigo resource", func() {
+				// Instantiating Lumigo after the deployment, so that the former is instrumented without the webhook
+				lumigo = newLumigo(namespaceName, "lumigo1", operatorv1alpha1.Credentials{
+					SecretRef: operatorv1alpha1.KubernetesSecretRef{
+						Name: lumigoSecretName,
+						Key:  expectedTokenKey,
+					},
+				}, true, true, true)
```

Comment:
```suggestion
				}, true, true, true)
```
those should be true by default right?

Reply:
Go does not really support overloading unfortunately